### PR TITLE
J sv1

### PR DESCRIPTION
**Pull request by jhallman5**
The console.log on line 31 is really helpful to see what is going on so I left it in the code for the purr review. My event listener on line 13 seems a bit hacky, especially lines 27 - 28. Functionalities which I still need to work on are the +/- , making the A/C => C transition, and %. Also I tried to get normalize.css via my package.json but I could not @import it into my CSS and eventually had to just create a file called normalize.css. Is this necessary?

### CODE REVIEW

**Comment by jhallman5** [comment]
> currently too many if conditionals. and my if conditionals in my numberCounter function do not operate as I expect. Is this a viable way to proceed overall? Calculator does not do proper order of operations. I'm also curious as to why I cant set current things to variables and call them later (such as document.querySelector('.calculator-screen').innerHTML).

File: app.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+var memory = {
+  initial: [],
+  operation: undefined,
+  latter: []
+}
+
+function operationAdder(div) {
+  memory.operation = this.innerHTML
```
Comment (jaredatron) [inline app.js]: you should not be reading from the DOM. `x = node.innerHTML` is reading state from the DOM

File: app.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+function numberCounter(div) {
+  if( !memory.operation && (this.innerHTML !== 'AC' || '+/-' || '%' || '.' )) {
```
Comment (jaredatron) [inline app.js:11]: Again do not read from the DOM. Listen for click events and key presses and then turn that even into a state change of your Javascript Calculator object.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      memory.initial.push(this.innerHTML)
+      document.querySelector('.calculator-screen').innerHTML = memory.initial.join('')
```
Comment (jaredatron) [inline app.js:13]: you're duplicating this code `document.querySelector('.calculator-screen').innerHTML`. Consider making a method called `setDisplayValue` or something

File: app.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  }else if( memory.operation && (this.innerHTML !== 'AC' || '+/-' || '%' || '.' )) {
+      memory.latter.push(this.innerHTML)
```
Comment (jaredatron) [inline app.js:15]: formatting. indented too far

File: app.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      document.querySelector('.calculator-screen').innerHTML = memory.latter.join('')
+  }
+}
+
+document.querySelectorAll(".calculator-button").forEach(function(div) {
+  div.addEventListener('click', numberCounter )
```
Comment (jaredatron) [inline app.js:21]: you should not need two click handlers here.

File: app.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  div.addEventListener('click', function(){
+    if(div.innerHTML === "AC"){
```
Comment (jaredatron) [inline app.js]: you should use a classname instead of the `div.innerHTML` content to determine what button was pressed

Comment (jaredatron) [reply app.js]: also look into `event bubbling` and `event delegation`

File: app.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    }else if( (div.innerHTML === '.') && (memory.latter.length > 0) && (memory.latter.indexOf('.') === -1 ) ){
+      memory.latter.push('.')
+      document.querySelector('.calculator-screen').innerHTML = memory.latter.join('')
+    }
+  })
+})
+
+document.querySelectorAll(".operator").forEach(function(div) {
+  div.addEventListener('click', operationAdder )
+  div.removeEventListener('click', numberCounter )
```
Comment (jaredatron) [inline app.js:75]: Removing these means you didn't properly add `numberCounter` as an event listener.

File: normalize.css [hunk 8]
```diff
@@ -0,0 +1,461 @@
+/*! normalize.css v5.0.0 | MIT License | github.com/necolas/normalize.css */
```
Comment (jaredatron) [inline normalize.css:1]: there are at least 2 better ways to get `normalize.css` into your project. Either use a CDN or `npm`

File: style.css [hunk 9]
```diff
@@ -40,7 +41,7 @@ body {
   line-height: 60px;
 }
 
-.calculator .orange  {
+.calculator .operator {
```
Comment (jaredatron) [inline style.css:44]: `.operator` needs to be namespaced under `.calculator-` so maybe `.calculator-operator-button` ?

File: style.css [hunk 10]
```diff
@@ -50,8 +51,9 @@ body {
   border-bottom-left-radius: 10px;
 }
 
-.calculator .corner{
+.calculator .corner {
```
Comment (jaredatron) [inline style.css:54]: `.calculator-corner` ?

**Review by jaredatron — CHANGES_REQUESTED** [review]

**Pull request merged**